app/controllers/api/base_controller/renderer.rb

```diff
@@ -420,6 +420,8 @@ def expand_subcollection(json, sc, sctype, subresources)
             :expand_resources => @req.expand?(sc)
           }
           json.set! sc.to_s, collection_to_jbuilder(sc.to_sym, sctype, subresources, copts)
+        elsif subresources.kind_of?(Hash)
+          json.set!(sc, subresources)
```

Comment:
This will not normalize it like the regular GET /api/settings, so it will return passwords and such, we'd need json.set!(sc, normalize_hash(....))